tmt/steps/execute/__init__.py

```diff
@@ -261,5 +261,11 @@ def check_beakerlib(self, test):
         except AttributeError:
             self.debug(f"No result in '{beakerlib_results_file}'.", level=3)
             return tmt.Result(data, test.name)
-        data['result'] = result.lower()
+        # Beakerlib itself has to end with 0
+        if test.returncode != 0:
+            data['result'] = 'error'
+            if test.returncode == tmt.utils.PROCESS_TIMEOUT:
```

Comment:
this could be little bit tricky,  or better to say, would be nice to cover with more of cases:

E.g. I'm not sure what are all possible return codes of beakerlib tests, to not confuse users e.g. if RC is 124 as defined for timeout, may hide another issue inside beakerlib output? But from my perspective 124 is high enough to not met this issue I think.

Reply:
124 (PROCESS_TIMEOUT) is set by tmt executor.
If you know any other case let me know - but it will affect only the text written to 'note'. 
Result will stay 'error'

Reply:
Nope, just ensure that we'll not cover something.

Reply:
@sopos @hegerj : Are there special beakerlib exit codes tmt should care about?

Reply:
No, even rlJournalEnd exit code does not correspond to the test result. Moreover, we cannot prevent user to put `exit 0` or `exit 1` at the end.

Beakerlib porvides the _TestResults_ file located in the $BEAKERLIB_DIR, where `TESTRESULT_RESULT_STRING` / `TESTRESULT_RESULT_ECODE` contains the current overall test result. Together with `TESTRESULT_STATE` set to `complete`, the `TESTRESULT_RESULT_STRING` / `TESTRESULT_RESULT_ECODE` indicate the final result.

Reply:
I see. Thank you @sopos . 
I'll change `check_beakerlib` to make sure `TESTRESULT_STATE` is complete.